compiler/codegen/stmt.cpp

```diff
@@ -95,6 +95,40 @@ GenRet UseStmt::codegen() {
 *                                                                   *
 ********************************* | ********************************/
 
+#ifdef HAVE_LLVM
+static
+void codegenLifetimeStart(llvm::Type *valType, llvm::Value *addr)
+{
+  GenInfo *info = gGenInfo;
+  const llvm::DataLayout& dataLayout = info->module->getDataLayout();
+
+  int64_t sizeInBytes = -1;
+  if (valType->isSized())
+    sizeInBytes = dataLayout.getTypeSizeInBits(valType)/8;
```

Comment:
Instead of `getTypeSizeInBits` divided by 8, this should be `getTypeStoreSize` not divided by 8.  What we're doing is marking memory, so we want the memory covered by the object (including any holes) to be marked.  Clang also uses this philosophy.